common/web/keyboard-processor/src/keyboards/activeLayout.ts

```diff
@@ -212,6 +233,51 @@ class ActiveKeyBase {
       rawKey.sp = Number.parseInt(rawKey.sp, 10) as ButtonClass;
     }
     rawKey.sp ||= 0; // The default button class.
+
+    // And now for generalized type validation. -----------------------------------------
+
+    // Object.entries does require Android 54... but we do polyfill within the Android app.  Should be 'fine'.
```

Comment:
destructuring is Chrome 49. So ... hope that is also polyfilled!

Reply:
Destructuring is TS-down-compiled, so no worries there.  This isn't the first time it's been used in the code base.

From the actual TS compilation:

```javascript
        // And now for generalized type validation. -----------------------------------------
        // Object.entries does require Android 54... but we do polyfill within the Android app.  Should be 'fine'.
        for (var _i = 0, _a = Object.entries(KeyTypesOfKeyMap); _i < _a.length; _i++) {
            var _b = _a[_i], key = _b[0], value = _b[1];
            switch (value) {
```

No destructuring syntax is in the final product.